From Larry.Finger@lwfinger.net  Wed Feb 17 15:45:53 2010
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Thu, 11 Feb 2010 14:41:24 -0600
Subject: Staging: r8187se: Fix for LED not following radio switch
To: Greg Kroah-Hartman <gregkh@suse.de>
Cc: Bernhard Schiffner <bernhard@schiffner-limbach.de>, linux-kernel@vger.kernel.org, linux-wireless@vger.kernel.org
Message-ID: <4b746b74.8SETcYq7pcTkXOg9%Larry.Finger@lwfinger.net>


The current driver does not follow the state of the RF switch.

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/rtl8187se/r8180_core.c |    7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

--- a/drivers/staging/rtl8187se/r8180_core.c
+++ b/drivers/staging/rtl8187se/r8180_core.c
@@ -4238,11 +4238,12 @@ void GPIOChangeRFWorkItemCallBack(struct
 	/* HW radio On/Off according to the value of FF51[4](config0) */
 	btConfig0 = btPSR = read_nic_byte(dev, CONFIG0);
 
-	/* Turn on LED. */
-	write_nic_byte(dev, PSR, btPSR | BIT3);
-
 	eRfPowerStateToSet = (btConfig0 & BIT4) ?  eRfOn : eRfOff;
 
+	/* Turn LED back on when radio enabled */
+	if (eRfPowerStateToSet == eRfOn)
+		write_nic_byte(dev, PSR, btPSR | BIT3);
+
 	if ((priv->ieee80211->bHwRadioOff == true) &&
 	   (eRfPowerStateToSet == eRfOn)) {
 		priv->ieee80211->bHwRadioOff = false;
